driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -356,6 +359,21 @@ public CreateCollectionOperation timeSeriesOptions(@Nullable final TimeSeriesOpt
         return this;
     }
 
+    public CreateCollectionOperation clusteredIndexKey(final BsonDocument clusteredIndexKey) {
+        this.clusteredIndexKey = clusteredIndexKey;
+        return this;
+    }
+
+    public CreateCollectionOperation clusteredIndexUnique(final boolean clusteredIndexUnique) {
+        this.clusteredIndexUnique = clusteredIndexUnique;
+        return this;
+    }
+
+    public CreateCollectionOperation clusteredIndexName(final String clusteredIndexName) {
```

Comment:
Let's mark `clusteredIndexName` as `@Nullable`, because we indeed may pass `null` when calling this method in `MongoOperationPublisher`.